packages/amplify-category-geo/src/service-walkthroughs/geofenceCollectionWalkthrough.ts

```diff
@@ -0,0 +1,246 @@
+      if (showAdvancedSettings) {
+        parameters = merge(parameters, await geofenceCollectionPricingPlanWalkthrough(context, parameters));
+      }
+  }
+
+  // ask if the geofence collection should be set as a default. Default to true if it's the only geofence collection
+  const currentGeofenceCollectionResources = await getGeoServiceMeta(ServiceName.GeofenceCollection);
+  if (currentGeofenceCollectionResources && Object.keys(currentGeofenceCollectionResources).length > 0) {
+    parameters.isDefault = await prompter.yesOrNo(
+        defaultResourceQuestion(ServiceName.GeofenceCollection),
+        true
+    );
+  }
+  else {
+    parameters.isDefault = true;
+  }
+
+  return parameters;
+};
+
+export const geofenceCollectionNameWalkthrough = async (context: any): Promise<Partial<GeofenceCollectionParameters>> => {
+    let collectionName;
+    while(!collectionName) {
+        const [shortId] = uuid().split('-');
+        const collectionNameInput = await prompter.input(
+            'Provide a name for the Geofence Collection:',
+            { validate: alphanumeric(), initial: `geofenceCollection${shortId}` }
+        );
+        if (await checkGeoResourceExists(collectionNameInput)) {
+            printer.info(`Geo resource ${collectionNameInput} already exists. Choose another name.`);
+        }
+        else collectionName = collectionNameInput;
+    }
+    return { name: collectionName };
+};
+
+export const geofenceCollectionAccessWalkthrough = async (
+    context: $TSContext,
+    parameters: Partial<GeofenceCollectionParameters>
+): Promise<Partial<GeofenceCollectionParameters>> => {
+    parameters.accessType = AccessType.CognitoGroups;
+    const userPoolGroupList = context.amplify.getUserPoolGroupList();
+    if (userPoolGroupList.length <= 0) {
+        printer.error('No Cognito groups exist in the project. Please add a Cognito group using "amplify update auth" and selecting "Create or update Cognito user pool groups"');
+        throw new Error('Failed to setup a Geofence Collection. Requires a Cognito group for Admin only access control settings');
+    }
+
+    let defaultSelectedGroups: string[] = [];
+
+    if (parameters.groupPermissions) {
+      defaultSelectedGroups = Object.keys(parameters.groupPermissions);
+    }
+
+    const selectedUserPoolGroups = await prompter.pick<'many', string>(
+        'Select one or more cognito groups to give access:',
+        userPoolGroupList,
+        { returnSize: 'many', initial: byValues(defaultSelectedGroups) }
+    );
+
+    let selectedUserPoolGroupsList: string[] = [];
+
+    if(typeof selectedUserPoolGroups === 'string') {
+        selectedUserPoolGroupsList = [selectedUserPoolGroups];
+    }
+    else {
+        selectedUserPoolGroupsList = selectedUserPoolGroups;
+    }
+
+    const groupCrudPermissionsFlow = async (group: string, defaults: string[] = []) => {
+        const possibleOperations = Object.keys(crudPermissionsMap).map(el => ({ name: el, value: el }));
```

Comment:
if you want the prompter to return the values instead, you could use:
```suggestion
        const possibleOperations = Object.entries(crudPermissionsMap).map(([name, value]) => ({ name, value}));
```